cmd/provenanced/cmd/genaccounts.go

```diff
@@ -640,3 +690,188 @@ func checkMsgTypeValid(registry types.InterfaceRegistry, msgTypeURL string) erro
 	}
 	return err
 }
+
+// AddGenesisDefaultMarketCmd returns add-genesis-default-market cobra command.
+func AddGenesisDefaultMarketCmd(defaultNodeHome string) *cobra.Command {
+	cmd := &cobra.Command{
+		Use: `add-genesis-default-market [--denom <denom>]
+
+All base accounts already in the genesis file will be given all permissions on this market.
+An error is returned if no such accounts are in the existing genesis file.
+
+If --denom <denom> is not provided, the staking bond denom is used.
+If no staking bond denom is defined either, then nhash is used.
+
+This command is equivalent to the following command:
+$ ` + version.AppName + ` add-genesis-custom-market \
+    --name 'Default <denom> Market' \
+    --create-ask 100<denom> --create-bid 100<denom> \
+    --seller-flat 500<denom> --buyer-flat 500<denom> \
+    --seller-ratios 20<denom>:1<denom> --buyer-ratios 20<denom>:1<denom> \
+    --accepting-orders --allow-user-settle \
+	--access-grants <all permissions to all known base accounts>
+
+`,
+		Short:                 "Add a default market to the genesis file",
+		DisableFlagsInUseLine: true,
+		Args:                  cobra.NoArgs,
+		RunE: updateGenesisFileRunE(func(clientCtx client.Context, cmd *cobra.Command, _ []string, appState map[string]json.RawMessage) error {
+			// Printing usage with the errors from here won't actually help anyone.
+			cmd.SilenceUsage = true
+
+			// Identify the accounts that will get all the permissions.
+			var authGen authtypes.GenesisState
+			err := clientCtx.Codec.UnmarshalJSON(appState[authtypes.ModuleName], &authGen)
+			if err != nil {
+				return fmt.Errorf("could not extract auth genesis state: %w", err)
+			}
+			genAccts, err := authtypes.UnpackAccounts(authGen.Accounts)
+			if err != nil {
+				return fmt.Errorf("could not unpack genesis acounts: %w", err)
+			}
+			addrs := make([]string, 0, len(genAccts))
+			for _, acct := range genAccts {
+				// We specifically only want accounts that are base accounts (i.e. no markers or others).
+				// This should also include the validator accounts (that have already been added).
+				baseAcct, ok := acct.(*authtypes.BaseAccount)
+				if ok {
+					addrs = append(addrs, baseAcct.Address)
+				}
+			}
+			if len(addrs) == 0 {
+				return errors.New("genesis file must have one or more BaseAccount before a default market can be added")
+			}
+
+			// Identify the denom that we'll use for the fees.
+			feeDenom, err := cmd.Flags().GetString(flagDenom)
+			if err != nil {
+				return fmt.Errorf("error reading --%s value: %w", flagDenom, err)
+			}
+			if len(feeDenom) == 0 {
+				var stGen stakingtypes.GenesisState
+				err = clientCtx.Codec.UnmarshalJSON(appState[stakingtypes.ModuleName], &stGen)
+				if err == nil {
+					feeDenom = stGen.Params.BondDenom
+				}
+			}
+			if len(feeDenom) == 0 {
```

Comment:
Does this branch condition need to be evaluated twice? Can probably move `feeDenom = "nhash"` up to line `751` to achieve the same effect.

Reply:
If I put it on line 751, then, when `stGen.Params.BondDenom` is an empty string, `feeDenom` would end up empty too. To account for that, there'd need to be a `len` check on it before setting `feeDenom` from it anyway.

The nice thing about having it like this, is that it puts things in priority: "Look here for the thing. If it wasn't there, look over there for it. If we still don't have it, use this default." If I moved it up to line 751, it'd mix together the last two parts: "Look here for the thing. If it wasn't there, use this default, but overwrite it with what's over there if that's got something."